### ISSUE TITLE
Time zone dependant timestamps

### ISSUE THREAD

**Issue by D4M5** [powercord-community/suggestions#185]
### Description
The plugin would allow a user to locally store information on a user ID's time zone and then would display the time of their message in their respective time zone next to the local user's time zone.

### More info
This plugin would be really useful for telling what time it is for a friend or any other user without having to calculate it and/or ask them every time, and the data about the time zones per user ID would be stored locally. A quick mock up of the idea has been provided below.
![image](https://user-images.githubusercontent.com/33693795/93228574-eac16a80-f775-11ea-9816-e03b3cc7fb02.png)

----

 - [x] I checked if this request wasn't already proposed
 - [x] I made sure this request follows Powercord's [plugin guidelines](https://powercord.dev/guidelines)

**Comment by ezist** [comment]
Working on this, this is my first time making a powercord plugin so don't expect much.

**Comment by ezist** [comment]
Nevermind. I need to learn about how to correctly make plugins first. If you want to look at what I built to try to work on it, I put it into [a gist](https://gist.github.com/ezist/0b3a7094709000a2b95eafb890ecaac7).

**Comment by D4M5** [comment]
It's fine, thanks for showing interest and attempting making it though!

**Comment by taikutsunan** [comment]
Hmm... I might take this up.

**Comment by taikutsunan** [comment]
Alright, I'll take this one up.
I'm not home yet, so I can't start immedately, but I'm definitely going to start sometime today.

**Comment by taikutsunan** [comment]
I'm having a bit of a problem now...
We would need to access the message author object/id to test for anything, correct? The MessageTimestamp object doesn't have any information about the user (which, seems reasonable), and I can't find anything relating to MessageTimestamp in the message module's object itself. I'm a bit stuck at this point.
I've only made one plugin that isn't itself that complex, so I probably just don't have the experience yet to tell what's needed.

**Comment by taikutsunan** [comment]
I'll have to drop this one too, I don't know any good solutions for this problem at the skill level I'm at as of now.

**Comment by Cr3atable** [comment]
@taikutsunan I don't think you can get the timezone of any users via Discord's API or anything like that. Maybe have it so you can set timezones on a per-user basis? I find the plugin would still be useful that way.

**Comment by D4M5** [comment]
Just going to quickly note that it is exactly what I meant for the plugin to be. Locally stored time zone data for users, would still be very useful

**Comment by MulverineX** [comment]
@Cr3atable well obviously it would be usually manual. But pre-nuke we discussed options for making it automatic. Such as integrating with discord.bio, scanning their nickname for a `(UTC-3)`, etc. 

Also a tooltip over recognized dates in a message that use stored info as an additional feature would be neat.

**Comment by aspenluxxxy** [comment]
I'm looking into this :)

**Comment by jeffboy70** [comment]
Bump!

**Comment by MulverineX** [comment]
@jeffboy70 
> Please avoid "bumping" issues by posting "+1", or asking if someone's willing to work on it. It can be frustrating to wait for an issue to complete, but it is even more frustrating to send junk emails to everyone (and make Staff' triage work harder!).

[readme](https://github.com/powercord-community/suggestions#i-really-want-x-plugin-but-i-cant-code)
